File: daily/2025/3169_Count_Days_Without_Meetings.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int countDays(int days, vector<vector<int>>& meetings) {
        auto compare = [](vector<int>& a, vector<int>& b){
            if (a[0] != b[0]) return a[0] < b[0];
            else return a[1] < b[1];
        };
        sort(meetings.begin(), meetings.end(), compare);
        // for (auto i : meetings) cout << i[0] << " " << i[1] << endl;
        int n = meetings.size();
        int end = 0;
        int ans = 0;
        for (int i = 0; i < n; i++){
            int curr_start = meetings[i][0];
            int curr_end = meetings[i][1];
            if (curr_start > end) ans += curr_start - end - 1;
            end = max(end, curr_end);
        }
        if (days > end) ans += days - end;
        return ans;
    }
};

int main(){
    int n = 10;
    vector<vector<int>> meetings = {{5,7},{1,3},{9,10}};
    // int n = 5;
    // vector<vector<int>> meetings = {{2,4},{1,3}};
    // int n = 6;
    // vector<vector<int>> meetings = {{1,6}};
    // int n = 57;
    // vector<vector<int>> meetings = {{3,49},{23,44},{21,56},{26,55},{23,52},{2,9},{1,48},{3,31}};
    Solution S;
    cout << S.countDays(n, meetings) << endl;
    return 0;
}

File: daily_python/3169_Count_Days_Without_Meetings.py
import os
import sys
from typing import List

class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        n = len(meetings)
        meetings.sort()
        end = 0
        ans = 0
        for curr_start, curr_end in meetings:
            if curr_start > end:
                ans += curr_start - end - 1
            end = max(end, curr_end)
        if days > end:
            ans += days - end
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')

            days = int(lines[0].split('=')[1].strip()[:-1])
            meetings = [list(map(int, meeting.split(','))) for meeting in lines[1].split('=')[1].strip()[2:-3].split('},{')]
            dataset.append((days, meetings))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for days, meetings in dataset:
        results.append(solution.countDays(days, meetings))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')